apps/web/src/pages/Skills/components/SkillTable.tsx

```diff
@@ -1,37 +1,102 @@
-import React from "react";
-import { ColumnDef, createColumnHelper } from "@tanstack/react-table";
+import React, { useEffect, useMemo, useState } from "react";
+import {
+  ColumnDef,
+  PaginationState,
+  createColumnHelper,
+} from "@tanstack/react-table";
 import { useIntl } from "react-intl";
 import { OperationContext, useQuery } from "urql";
 import { useLocation } from "react-router-dom";
+import { SubmitHandler } from "react-hook-form";
 
 import { commonMessages, getLocalizedName } from "@gc-digital-talent/i18n";
 import { notEmpty, unpackMaybes } from "@gc-digital-talent/helpers";
 import { Pending } from "@gc-digital-talent/ui";
-import { Skill, graphql } from "@gc-digital-talent/graphql";
+import { Skill, SkillCategory, graphql } from "@gc-digital-talent/graphql";
 
 import useRoutes from "~/hooks/useRoutes";
 import Table from "~/components/Table/ResponsiveTable/ResponsiveTable";
 import cells from "~/components/Table/cells";
 import adminMessages from "~/messages/adminMessages";
 import { normalizedText } from "~/components/Table/sortingFns";
+import {
+  INITIAL_STATE,
+  SEARCH_PARAM_KEY,
+} from "~/components/Table/ResponsiveTable/constants";
 
 import {
   categoryAccessor,
   familiesAccessor,
   keywordsAccessor,
   skillFamiliesCell,
 } from "./tableHelpers";
+import SkillFilterDialog, { FormValues } from "./SkillFilterDialog";
+
+export type SkillFilterInput = FormValues;
+
+export function stringToEnumSkillCategory(
+  selection: string,
+): SkillCategory | undefined {
+  if (Object.values(SkillCategory).includes(selection as SkillCategory)) {
+    return selection as SkillCategory;
+  }
+  return undefined;
+}
+
+export function transformFormValuesToSkillFilterInput(
+  data: FormValues,
+): SkillFilterInput {
+  return {
+    skillCategories: data.skillCategories?.length
+      ? data.skillCategories.map(stringToEnumSkillCategory).filter(notEmpty)
+      : undefined,
+    skillFamilies: data.skillFamilies?.length ? data.skillFamilies : undefined,
+  };
+}
+
+export function transformSkillFilterInputToFormValues(
+  input: SkillFilterInput | undefined,
+): FormValues {
+  return {
+    skillCategories: input?.skillCategories?.filter(notEmpty) ?? [],
+    skillFamilies: input?.skillFamilies?.filter(notEmpty) ?? [],
+  };
+}
 
 const columnHelper = createColumnHelper<Skill>();
 
 interface SkillTableProps {
   skills: Array<Skill>;
   title: string;
+  paginationState?: PaginationState;
+  addButton?: boolean;
 }
 
-export const SkillTable = ({ skills, title }: SkillTableProps) => {
+export const SkillTable = ({
+  skills,
+  title,
+  paginationState,
+  addButton,
+}: SkillTableProps) => {
   const intl = useIntl();
   const paths = useRoutes();
+  const searchParams = new URLSearchParams(window.location.search);
```

Comment:
I think we usually use the React Router hook for search params, like https://github.com/GCTC-NTGC/gc-digital-talent/blob/66cec5433f1bb3bedbeffd1a8a3e2b7e6f3f22e4/apps/web/src/components/Table/ResponsiveTable/ResponsiveTable.tsx#L99 .

Reply:
Nice catch. Fixed by [ef7aa26](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9614/commits/ef7aa26bad0ad98c9600244852920e36f8cc30f7)